### PR TITLE
Feature/allow mac users to disable dock icon

### PR DESCRIPTION
**Pull request by mgmathus**
it's necesary to user the new runwar arg --dock-enable true | false

### CODE REVIEW

File: src/main/java/runwar/Server.java [hunk 1]
```diff
@@ -409,19 +408,23 @@ public synchronized void startServer(final ServerOptions options) throws Excepti
...
+                    Class<?> appClass = Class.forName("com.apple.eawt.Application");
+                    Method getAppMethod = appClass.getMethod("getApplication");
+                    Object appInstance = getAppMethod.invoke(null);
+                    Method dockMethod = appInstance.getClass().getMethod("setDockIconImage", java.awt.Image.class);
+                    dockMethod.invoke(appInstance, dockIcon);
+                } catch (Exception e) {
+                    LOG.warn("error setting dock icon image", e);
+                }
+            } else {
+                System.setProperty("apple.awt.UIElement", "true");
```
Comment (bdw429s) [inline src/main/java/runwar/Server.java:427]: @mgmathus I'm curious, what does this line of code do here when the doc icon is disabled?

**Comment by bdw429s** [comment]
@mgmathus This looks great.  Would you like to add the changes to CommandBox to use this flag now?  Should all be contained to the ServerService and you can follow another flag like the tray enable to just copy what it does.

File: src/main/java/runwar/Server.java (hunk 1 above)
Comment (mgmathus) [reply src/main/java/runwar/Server.java:427]: @bdw429s by default, when a java app is launched on Mac OS, there is a coffee cup icon that pops in the dock, what Denny Did with that portion of code is customize that icon with the one used in the tray icon, so the dock icon is not the generic one,

The dock icon is been removed by line 427

Comment (bdw429s) [reply src/main/java/runwar/Server.java:427]: Ahh, makes sense.  Thanks for the explanation!

**Comment by mgmathus** [comment]
ok, I will do it

**Pull request merged**